Intermediate.ATX.2017Q1.Cruz.Mendez

cruzmndz opened the pull request:
# Checkpoint Rubric

This is the rubric that your instructor will use to grade your checkpoints. Please do not edit. 

## Checkpoint 1: Towers of Hanoi Terminal App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [ ] Properly indented code: 15 points
- [x] Demonstrated effective use of JavaScript: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*






## Checkpoint 2: Towers of Hanoi GUI App
- [x] All tests passed: 55 points
- [x] Properly commented code: 15 points
- [x] Properly indented code: 15 points
- [x] Demonstrated effective use of jQuery: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*






## Checkpoint 3: Final Project Gist Blog
- [ ] Effective use of AJAX: 20 points
- [ ] Effective use of loops: 20 points
- [ ] Effective use of github API: 20 points
- [ ] Style and design: 10 points
- [ ] Properly commented code: 15 points
- [ ] Properly indented code: 15 points

Specific Comments:

*Comments may also be made on the file itself using github's commenting tools.*

cruzmndz committed:
Lesson Zero Testing

cruzmndz committed:
L1

cruzmndz committed:
L1 - all tests passing

forgot to commit

cruzmndz committed:
RPS app

nevetsretlaw commented on apps/02RockPaperScissors.js:
```diff
@@ -10,7 +10,30 @@ var rl = readline.createInterface({
 function rockPaperScissors(hand1, hand2) {
 
-  // Write code here
+if (hand1 === hand2) {
```
Comments will help explain what is going on in the code

cruzmndz committed:
RPS app

jeffgojeff commented on apps/02RockPaperScissors.js:
```diff
@@ -10,7 +10,30 @@ var rl = readline.createInterface({
+   return 'It\'s a tie!';
+  }
+
+if (hand1 === 'rock') {
```
add .toLowercase() to the end of 'hand1' and 'hand2' to potentially deter a user error

jeffgojeff commented on apps/02RockPaperScissors.js:
```diff
@@ -10,7 +10,30 @@ var rl = readline.createInterface({
+    return 'Hand one wins!';
+  }
+    return 'Hand two wins!'
+}
```
missing else statement

jeffgojeff commented on apps/02RockPaperScissors.js:
```diff
@@ -10,7 +10,30 @@ var rl = readline.createInterface({
+  if (hand2 === 'scissors') {
+    return 'Hand two wins!';
+  }
+    return 'Hand one wins!';
```
missing else statement

cruzmndz committed:
final

added .toLowerCase

cruzmndz committed:
all passing

struggled most on problem 4

cruzmndz committed:
PigLatin

all tests passing. need to redo code for test #2.

cruzmndz committed:
Homework 3

all tests passing

cruzmndz committed:
TicTacToe

all tests passing. Still need to add test to check for spot availability

cruzmndz committed:
All tests passing

cruzmndz committed:
All tests good

Dun.

cruzmndz committed:
Tests passing

homework five

pieroapretto commented on homework/05LessonFive.js:
```diff
@@ -63,15 +63,15 @@ for (forLoopCounter = 0; forLoopCounter > -10; forLoopCounter--) {
 //
 // What are loops? Why do we use them?
-//
+//LOOPS ARE STATEMENTS THAT RUN CONTINUOUSLY AS LONG AS A SPECIFIED CONDITION IS TRUE.
```
CORRECT.

pieroapretto commented on apps/03TicTacToe.js:
```diff
@@ -24,25 +24,57 @@ function printBoard() {
+    console.log('Player ' + playerTurn + ' Won!');
+    console.log('Player ' + playerTurn + ' Won!');
+    console.log('Player ' + playerTurn + ' Won!');
+    console.log('WAY TO GO PLAYER ' + playerTurn + '!');
```
![YES](http://38.media.tumblr.com/e85555e07fecee34e81d875ef534ce3e/tumblr_inline_nt0dkb9BlG1qkhtip_500.gif)

pieroapretto commented on apps/03TicTacToe.js:
```diff
@@ -24,25 +24,57 @@ function printBoard() {
+    console.log('WAY TO GO PLAYER ' + playerTurn + '!');
+//SHOW THE WINNING BOARD
+    printBoard();
+//RESET THE BOARD FOR THE NEXT GAME
```
Good use of comments here.

pieroapretto commented on apps/03TicTacToe.js:
```diff
@@ -24,25 +24,57 @@ function printBoard() {
+//PRINT TO BOARD
+  board[row][column] = playerTurn;
+//DETERMIN IF THERE IS A WINNER
+  checkForWin()
```
missing semicolon here.

pieroapretto commented on apps/04TowersOfHanoi.js:
```diff
@@ -12,31 +12,52 @@ var stacks = {
-
+  if (isLegal(startStack, endStack)) {
+    movePiece(startStack, endStack);
+      checkForWin();
```
All your logic checks out for this application. Good job.

pieroapretto commented on homework/04LessonFour.js:
```diff
@@ -77,7 +86,7 @@ var nestedObjects = {
 };
 // update the starring property to have the value 'Tim Robbins'
-
+nestedObjects.users['bob roberts'].starring = 'Tim Robbins';
```
Nicely done.

![dance](https://media.giphy.com/media/l46CdUTy21h8wwrAs/giphy.gif)

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -12,31 +12,52 @@ var stacks = {
+    return true;
+  } else {
+      console.log('Ahh Ahh Ah, you didnt say the magic word!!')
+        return false;
```
Make sure everything within a group of curly braces is aligned together, even return statements.

cruzmndz committed:
Mastermind tests passing

still need to play

cruzmndz committed:
syntax fix

moved if else bracket

cruzmndz committed:
fixed syntax errors

moved indented objects inward.

cruzmndz committed:
removed line 62

return board, instead of 'true'

cruzmndz committed:
HMWK 6 - done

tests passing

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -12,31 +12,52 @@ var stacks = {
+  if (tower1[tower1.length - 1] < tower2[tower2.length-1]) {
+    return true;
+  } else {
+      console.log('Ahh Ahh Ah, you didnt say the magic word!!')
```
I forget what my previous comment was but make sure you are indenting everything from the left margin not the word of the else
so in this case you have indented 2 spaces from the "e" but you should be indenting base on the left margin of the entire file. 
so line 40 and line 42 should be indented the same

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -12,31 +12,52 @@ var stacks = {
+    console.log('ERES EL GANADORRRRR!!!');
+    return true;
+  } else {
+      return false;
```
same thing here, line 50 and 52 should be lined up

jw56578 commented on apps/04TowersOfHanoi.js:
```diff
@@ -12,31 +12,52 @@ var stacks = {
+  if ((stacks.b.length === 4) || (stacks.c.length === 4)) {
+    console.log('ERES EL GANADORRRRR!!!');
+    return true;
+  } else {
```
I would prefer if you broke your else down to the next line

cruzmndz committed:
first commit

One more test needed to pass

cruzmndz committed:
Final

all tests passing

cruzmndz committed:
First push (top horizontal win only)

need to add all the other winning methods.

cruzmndz committed:
final

all tests passing plus bonus spec.

cruzmndz committed:
first

two tests passing

cruzmndz committed:
Final

all tests passing. added push checkers to count checkers in play.

pieroapretto commented on apps/05Mastermind.js:
```diff
@@ -29,12 +29,37 @@ function getRandomInt(min, max) {
 
 function mastermind(guess) {
-  // your code here
+  solution = 'abcd';
```
You can remove the predefined solution after you're done with testing.

pieroapretto commented on apps/07jquery-tic-tac-toe/script.js:
```diff
@@ -1,5 +1,81 @@
+      getElement('[data-cell="8"]').text() === playerTurn) {
+        return true;
+  }
+  //DIAGONAL WINS
```
Good job leaving comments when using multiple `if` statements.

pieroapretto commented on apps/06SpaceTravelToMars.js:
```diff
@@ -10,6 +10,46 @@ var jobTypes = {
+//ADDED THIS METHOD PER THE SPECS TO COMPARE SHIP'S TYPE AND CREW'S JOBS
+    this.missionStatement = function() {
+      for (var i = 0; i < this.crew.length; i++) {
+        if (jobTypes[this.crew[i].job] === this.type) {
```
Nicely done.

cruzmndz committed:
first push

blocks can move now.

pieroapretto commented on homework/06LessonSix.js:
```diff
@@ -76,27 +81,31 @@ function ConferenceRoom() {
 //
 // What is a method?
-//
+//A METHOD IS A TOOL/FUNCTION USED TO MODIFY DATA IN AN OBJECT.
```
Yup. It's a function assigned to an object as an object property.

cruzmndz committed:
complete

3 tests passing

cruzmndz committed:
complete?

issues with npm?

cruzmndz committed:
update

outdated command prompt?

cruzmndz committed:
all tests passing

+ updated HTML & CSS for proper stacking.

pieroapretto commented on apps/08jquery-towers-of-hanoi/script.js:
```diff
@@ -1,20 +1,21 @@
 function stackWasClicked() {
   //IF BLOCK HAS SOMETHING IN IT WE ALREADY PICKED UP A BLOCK. 
-  if (block) {    //ADDED  !== NULL) {
+  if (block !== null) {
```
You can also pass `if(block) { ... ` for short

pieroapretto commented on homework/08LessonEight/script.js:
```diff
@@ -17,30 +17,30 @@ $(document).on('ready', function() {
   // use $finalList.append() to put window.secondListChildren
   // at the end of #final-list
-  $finalList;
+  $finalList.append(window.secondListChildren);
```
All tests passed! You're too smart for this class.

![passing](https://media.giphy.com/media/jBLJuTezU5tra/giphy.gif)

cruzmndz committed:
first commit

spec 1 and 2 done.

cruzmndz committed:
final

-removed .on('click', and moved up function call to use .click(function
-commented out place holders in html

cruzmndz committed:
first commit

halfway through spec 2.

pieroapretto commented on apps/10address-book-ajax/script.js:
```diff
@@ -1,5 +1,44 @@
+                  "</tr>";
+        $('tbody').append(str);
+      })  
+      viewUser();  
```
Good job calling this event listener function after your success callback.

pieroapretto commented on apps/10address-book-ajax/script.js:
```diff
@@ -1,5 +1,44 @@
+    console.log('user was clicked');
+    $.ajax(url, {
+      success: function(userArray) {
+        var viewStr = "<h3>" + userArray.first_name + "</h3>" +
```
Missing `last_name` here.

pieroapretto commented on apps/11gist-blog/script.js:
```diff
@@ -1,5 +1,30 @@
 
 $(document).ready(function() {
-  // You code here
+  $.ajax('http://127.0.0.1:8080/apps/11gist-blog/api/gists.json', {
```
You can shorten this ajax call to `$.ajax("api/gists.json",` since the api folder is relatively close to the js file. Very similar to how you handle linking css files to an html page.

pieroapretto commented on apps/11gist-blog/script.js:
```diff
@@ -1,5 +1,30 @@
+      userPosts.forEach(function(userGist) {
+        var userSubstr = userGist.description; 
+        var postString = "<li>" + 
+                         "<a href='#' data-url='" + userGist.url + "'>" + userGist.description.substring(6,userSubstr.length) + "</a>" + 
```
Nicely done scrubbing your description. You can also pass in just the starting spot to the substring method and it will extend to length of the string. 

ex: `userGist.description.substring(6) + ..` here

pieroapretto commented on apps/11gist-blog/script.js:
```diff
@@ -1,5 +1,30 @@
+    success: function(userPosts) {
+      userPosts.forEach(function(userGist) {
+        var userSubstr = userGist.description; 
+        var postString = "<li>" + 
```
Only 2 posts should pass through; the 2 that start with `"#post "`. Use an `if` statement to filter your posts.

pieroapretto commented on apps/11gist-blog/script.js:
```diff
@@ -1,5 +1,30 @@
+    console.log('url was clicked');   
+    $.ajax(gistURL, {
+      success: function(gistContent) {
+        var postText = "<p>" + gistContent.content + "</p>";
```
Getting the appropriate comments requires some serious digging. 

Refer to your 04LessonFour assignment for how to access and manipulate objects.

Ex: `nestedObjects.users['bob roberts'].starring = 'Tim Robbins';`

cruzmndz committed:
blog

posts and comments up. Minimal CSS
